src/Components/Admin/ObjectEditor/keypadpuzzle.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from "react";
+import AddIcon from "@material-ui/icons/Add";
+import { Button, IconButton } from "@material-ui/core";
+import { DeleteForever } from "@material-ui/icons";
+import TextField from "@material-ui/core/TextField";
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(() => ({
+    input: {
+        "&:invalid": {
+            border: "red solid 2px",
+        },
+    },
+}));
+
+export default function KeypadPuzzle(props) {
+    const classes = useStyles();
+    const [tempPass, setTempPass] = useState(props.pass);
+    const [pass, setPass] = useState(props.pass);
+    const [showNumpadPrompt, setNumpadPrompt] = useState(false);
+    const [showKeyboardPrompt, setKeyboardPrompt] = useState(false);
+    const [showSavePass, setSavePass] = useState(false);
+
+    const showPrompt = () => {
+        if (props.isNumpad) {
+            setNumpadPrompt(true);
+        } else {
+            setKeyboardPrompt(true);
+        }
+        setSavePass(true);
+    };
+
+    const deletePass = () => {
+        props.savePass("");
+        setPass("");
+        if (props.isNumpad) {
+            setNumpadPrompt(false);
+        } else {
+            setKeyboardPrompt(false);
+        }
+        setSavePass(false);
+    };
+
+    const savePass = () => {
+        props.savePass(tempPass);
+        setPass(tempPass);
+    };
+
+    const numpadOnChange = (e) => {
+        const re = /^[0-9\b]{1,7}$/;
+        if (e.target.value === "" || re.test(e.target.value)) {
+            setTempPass(e.target.value);
+            setSavePass(true);
+        } else {
+            alert(
+                "Error: Password must be at most 7 characters and only consist of numbers"
+            );
+            setSavePass(false);
+        }
+    };
+
+    const keyboardOnChange = (e) => {
+        const re = /^[0-9a-zA-Z\b]{1,7}$/;
```

Comment:
I think keyboard password can be longer than 7 characters (I see no spacial reason on the keyboard interface to indicate otherwise). However when testing currently, I was only allowed to enter 5 (likely because expected password is "abcde" which is 5 characters). We should probably allow users to input more than the expected password length, if that is possible to do. @jaydhulia or @drbalar can you verify if that is possible or not?

Image:
![image](https://user-images.githubusercontent.com/13268990/134785687-0d459eb0-b6a8-4f98-910d-d1fad3ede317.png)

Reply:
Yea it is possible to remove the password length restriction

Reply:
Set max length to 16 as per recommendation